pkg/config/config.go

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package config - contain environment variables used by nsc
+package config
+
+import (
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/networkservicemesh/api/pkg/api/networkservice/mechanisms/kernel"
+	"github.com/networkservicemesh/api/pkg/api/networkservice/mechanisms/memif"
+)
+
+var configMechanisms = map[string]string{
+	"memif":  memif.MECHANISM,
+	"kernel": kernel.MECHANISM,
+}
+
+// Config - configuration for cmd-nsmgr
+type Config struct {
+	Name             string        `default:"nsc" desc:"Name of Network Service Client"`
+	ConnectTo        *url.URL      `default:"[unix:///var/lib/networkservicemesh/nsm.io.sock" desc:"url to connect to NSM" split_words:"true"`
+	MaxTokenLifetime time.Duration `default:"24h" desc:"maximum lifetime of tokens" split_words:"true"`
+
+	Routes    []string `default:"" desc:"A list of routes asked by client" split_words:"true"`
+	Labels    []string `default:"" desc:"A list of client labels with format key1=val1,key2=val2, will be used a primary list for network services" split_words:"true"`
+	Mechanism string   `default:"kernel" desc:"Default Mechanism to use, supported values kernel,memif" split_words:"true"`
+
+	NetworkServices []*NetworkServiceConfig `default:"" desc:"A list of Network Service Requests with format [{mechanism}]?:${nsName}[@domainName]?/${interfaceName/memIfSocketName}?${label1}=${value1}&${label2}=${value2}" split_words:"true"`
+}
+
+// IsValid - check if configuration is valid
+func (c *Config) IsValid() error {
+	if len(c.NetworkServices) == 0 {
+		return errors.New("no network services are specified")
+	}
+	if c.Name == "" {
+		return errors.New("no cleint name specified")
+	}
+	if c.ConnectTo == nil {
+		return errors.New("no NSMGr ConnectTO URL are sepecified")
+	}
+	return nil
+}
+
+// NetworkServiceConfig - defines a network service request configuration
+type NetworkServiceConfig struct {
+	NetworkService string            `default:"" desc:"A name of network service" split_words:"true"`
```

Comment:
Can the client want a few network services?
```suggestion
	NetworkService []string            `default:"" desc:"A name of network service" split_words:"true"`
```